fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/popcha/PoPCHAHomeFragmentTest.java

```diff
@@ -0,0 +1,183 @@
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import com.github.dedis.popstellar.model.network.method.message.data.popcha.PoPCHAAuthentication;
+import com.github.dedis.popstellar.model.objects.*;
+import com.github.dedis.popstellar.model.objects.event.EventState;
+import com.github.dedis.popstellar.model.objects.security.*;
+import com.github.dedis.popstellar.model.objects.view.LaoView;
+import com.github.dedis.popstellar.repository.LAORepository;
+import com.github.dedis.popstellar.repository.RollCallRepository;
+import com.github.dedis.popstellar.repository.remote.GlobalNetworkManager;
+import com.github.dedis.popstellar.testutils.BundleBuilder;
+import com.github.dedis.popstellar.testutils.MessageSenderHelper;
+import com.github.dedis.popstellar.testutils.fragment.ActivityFragmentScenarioRule;
+import com.github.dedis.popstellar.testutils.pages.lao.LaoActivityPageObject;
+import com.github.dedis.popstellar.ui.lao.LaoActivity;
+import com.github.dedis.popstellar.utility.Constants;
+import com.github.dedis.popstellar.utility.error.UnknownLaoException;
+import com.github.dedis.popstellar.utility.error.keys.KeyException;
+import com.github.dedis.popstellar.utility.security.KeyManager;
+import dagger.hilt.android.testing.*;
+import io.reactivex.subjects.BehaviorSubject;
+import java.util.HashSet;
+import javax.inject.Inject;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoTestRule;
+
+@HiltAndroidTest
+@RunWith(AndroidJUnit4.class)
+public class PoPCHAHomeFragmentTest {
+  private static final long CREATION_TIME = 1612204910;
+  private static final String LAO_NAME = "laoName";
+  private static final KeyPair SENDER_KEY = generateKeyPair();
+  private static final Lao LAO = new Lao(LAO_NAME, SENDER_KEY.getPublicKey(), CREATION_TIME);
+  private static final BehaviorSubject<LaoView> laoSubject =
+      BehaviorSubject.createDefault(new LaoView(LAO));
+  private static final String ROLLCALL_NAME = "rollcall#1";
+  private static final String ROLLCALL_ID =
+      RollCall.generateCreateRollCallId(LAO.getId(), CREATION_TIME, ROLLCALL_NAME);
+  private static final long TIMESTAMP_1 = 1632204910;
+  private static final long TIMESTAMP_2 = 1632204900;
+  private static final PoPToken popToken = generatePoPToken();
+  private static final AuthToken authToken = new AuthToken(popToken);
+  private static final HashSet<PublicKey> attendees = new HashSet<>();
+
+  static {
+    attendees.add(popToken.getPublicKey());
+  }
+
+  private static final RollCall ROLL_CALL =
+      new RollCall(
+          ROLLCALL_ID,
+          ROLLCALL_ID,
+          ROLLCALL_NAME,
+          CREATION_TIME,
+          TIMESTAMP_1,
+          TIMESTAMP_2,
+          EventState.CLOSED,
+          attendees,
+          "bc",
+          "");
```

Comment:
considering almost all the constants here are not (really) used in the test, but only in the setup, and I imagine (maybe mistakenly) they might be copied from other tests where they're similarly not really used:

Could this logic be put in a test utility class to generate this stuff for all tests ? 
and/or
Could these constants be removed and expressed as succinctly as reasonably possible in the test setup itself ?

Reply:
Yes, that could be improved 😄. I need to validate where in all the tests this is reused to refactor it

Reply:
Okay, looking better at the code refactoring this goes out of the scope of this PR as it involves a lot of changes, so I'll leave that for future improvements. Instead I will contain as possible the constants used here and express them in the setup